application-analytics-test/application-analytics-test-pageobjects/src/main/java/com/xwiki/analytics/test/po/MostViewedMacroViewPages.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.analytics.test.po;
+
+import java.util.List;
+import java.util.Arrays;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.xwiki.test.ui.XWikiWebDriver;
+import org.xwiki.test.ui.po.ViewPage;
+
+public class MostViewedMacroViewPages extends ViewPage
+{
+
+    private  static XWikiWebDriver driver;
+    private static final String ROW_EVOLUTION_BUTTON_SELECTOR = ".analyticsRowEvolution";
+    private static final String ROW_EVOLUTION_MODAL_SELECTOR = ".modal.fade.analyticsRowEvolutionModal";
+    public MostViewedMacroViewPages()
+    {
+        driver =  getUtil().getDriver();
+    }
+
+    public static MostViewedMacroViewPages gotoPage()
```

Comment:
Why is this and the rest of the following methods static? A page object is not a "Utils" class providing static utility methods for tests. We use page objects to model in an object-oriented way the UI that the user interacts with. The web page the user interacts with may have a header (top bar), a menu, a body, a footer. The body could include a form, which includes multiple fields. Two of these fields could be a rich text editor, which has a toolbar and a rich text area, so on. All these UI elements are usually modeled by separate page objects, allowing you to write tests that do:

    new RichTextEditor(somePage.getForm('foo').getField('bar')).getToolBar().click('Bold');